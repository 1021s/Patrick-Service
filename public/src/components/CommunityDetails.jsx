import React from 'react';
import ListHelper from './ListHelper';

const CommunityDetails = ({ listing }) => (
  <div>
    <div>
      <div className="type">Location</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.location).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.location[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default CommunityDetails;
