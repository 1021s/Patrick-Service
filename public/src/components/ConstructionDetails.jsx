/* eslint-disable react/prop-types */
import React from 'react';
import ListHelper from './ListHelper';

const ConstructionDetails = ({ listing }) => (
  <div className="construction-details-container">
    <div>
      <div className="type">Type and style</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.typeAndStyle).map((key) => (
          <ListHelper key={key} entry={key} value={listing.typeAndStyle[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Material information</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.materialInformation).map((key) => (
          <ListHelper key={key} entry={key} value={listing.materialInformation[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Condition</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.condition).map((key) => (
          <ListHelper key={key} entry={key} value={listing.condition[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default ConstructionDetails;
