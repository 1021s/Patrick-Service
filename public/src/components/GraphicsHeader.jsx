/* eslint-disable react/prop-types */
import React from 'react';
import '../../dist/styles.css';

const GraphicsHeader = ({ listing }) => (
  <div className="facts-container">
    <div className="graphics-container">
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/city-buildings.png" alt="" length="20px" width="20px" />
        &#8203; Type:
      </div>
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/calendar.png" alt="" length="20px" width="20px" />
        &#8203; Year built:
      </div>
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/thermometer.png" alt="" length="20px" width="20px" />
        &#8203; Heating:
      </div>
    </div>
    <div className="graphics-container">
      <div className="kind">{listing.type}</div>
      <div className="kind">{listing.yearBuilt}</div>
      <div className="kind">{listing.heating}</div>
    </div>
    <div className="graphics-container">
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/winter.png" alt="" length="20px" width="20px" />
        &#8203; Cooling:
      </div>
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/parking.png" alt="" length="20px" width="20px" />
        &#8203; Parking:
      </div>
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/floor-plan.png" alt="" length="20px" width="20px" />
        &#8203; Lot:
      </div>
      <div className="type">
        <img src="https://img.icons8.com/ultraviolet/40/000000/surface.png" alt="" length="20px" width="20px" />
        &#8203; Price/sqft:
      </div>
    </div>
    <div className="graphics-container">
      <div className="kind">{listing.cooling}</div>
      <div className="kind">{listing.parking}</div>
      <div className="kind">{listing.lot}</div>
      <div className="kind">{listing.priceSqft}</div>
    </div>
  </div>
);

export default GraphicsHeader;
