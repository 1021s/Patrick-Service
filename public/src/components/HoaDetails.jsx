import React from 'react';
import ListHelper from './ListHelper';

const HoaDetails = ({ listing }) => (
  <div>
    <div>
      <div className="type">Other financial information</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.otherFinancialInformation).map((key) => (
          <ListHelper key={key} entry={key} value={listing.otherFinancialInformation[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default HoaDetails;
