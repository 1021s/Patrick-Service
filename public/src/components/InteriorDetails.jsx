import React from 'react';
import ListHelper from './ListHelper';

const InteriorDetails = ({ listing }) => (
  <div>
    <div>
      <div className="type">Bedrooms and bathrooms</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.bedroomsAndBathrooms).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.bedroomsAndBathrooms[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Basement</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.basement).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.basement[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Flooring</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.flooring).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.flooring[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Heating</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.heating).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.heating[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Appliances</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.appliances).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.appliances[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Other interior features</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.otherInteriorFeatures).map((key, i) => (
          <ListHelper key={i} entry={key} value={listing.otherInteriorFeatures[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default InteriorDetails;
