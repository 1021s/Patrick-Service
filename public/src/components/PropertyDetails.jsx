/* eslint-disable react/prop-types */
import React from 'react';
import ListHelper from './ListHelper';

const PropertyDetails = ({ listing }) => (
  <div>
    <div>
      <div className="type">Parking</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.parking).map((key) => (
          <ListHelper key={key} entry={key} value={listing.parking[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Property</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.property).map((key) => (
          <ListHelper key={key} entry={key} value={listing.property[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Lot</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.lot).map((key) => (
          <ListHelper key={key} entry={key} value={listing.lot[key]} />
        ))}
      </ul>
    </div>
    <div>
      <div className="type">Other property information</div>
      <ul>
        {typeof listing === 'object' && Object.keys(listing.otherPropertyInformation).map((key) => (
          <ListHelper key={key} entry={key} value={listing.otherPropertyInformation[key]} />
        ))}
      </ul>
    </div>
  </div>
);

export default PropertyDetails;
