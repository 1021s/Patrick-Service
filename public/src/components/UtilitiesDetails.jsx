import React from 'react';

const UtilitiesDetails = ({ listing }) => (
  <div>

  </div>
);

export default UtilitiesDetails;
