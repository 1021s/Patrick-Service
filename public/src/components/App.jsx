/* eslint-disable consistent-return */
import React from 'react';
import axios from 'axios';
import GraphicsHeader from './GraphicsHeader';
import InteriorDetails from './InteriorDetails';
import PropertyDetails from './PropertyDetails';
import ConstructionDetails from './ConstructionDetails';
import UtilitiesDetails from './UtilitiesDetails';
import CommunityDetails from './CommunityDetails';
import HoaDetails from './HoaDetails';
import Other from './Other';
import Modal from './Modal';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      listing: {},
      expanded: false,
      modal: false,
      appClass: '',
    };
    this.componentDidMount = this.componentDidMount.bind(this);
    this.expand = this.expand.bind(this);
    this.toggleModal = this.toggleModal.bind(this);
  }

  componentDidMount() {
    const parsedUrl = new URL(window.location.href);
    axios.get(
      `/listings/${parsedUrl.href.slice(-3)}`,
    )
      .then((data) => {
        this.setState({
          listing: data.data[0],
        });
      })
      .catch((err) => err);
  }

  expand(event) {
    event.preventDefault();
    const { expanded } = this.state;
    if (expanded === false) {
      this.setState({
        expanded: true,
      });
    } else {
      this.setState({
        expanded: false,
      });
    }
  }

  toggleModal(event) {
    event.preventDefault();
    const { modal } = this.state;
    if (!modal) {
      this.setState({
        modal: !modal,
        appClass: 'app-modal',
      });
    } else {
      this.setState({
        modal: !modal,
        appClass: 'app-modal',
      });
    }
  }

  render() {
    const {
      listing, expanded, modal, appClass,
    } = this.state;
    if (expanded === false) {
      return (
        <div className="App">
          <h1 className="title-line">Facts and features</h1>
          <GraphicsHeader listing={listing} />
          <div className="link" onClick={this.expand} tabIndex={0} onKeyPress={this.expand} role="button">
            <img src="https://img.icons8.com/ultraviolet/40/000000/circled-chevron-down.png" alt="" height="20px" width="20px" />
            See more facts and features
          </div>
        </div>
      );
    }
    if (expanded === true && modal === true) {
      return (
        <div>
          <Modal show={modal} toggleModal={this.toggleModal} listing={listing} />
          <div className={appClass}>
            <h3 className="title-line">Facts and features</h3>
            <GraphicsHeader listing={listing} />
            <div className="details">Interior details</div>
            <InteriorDetails listing={listing.interiorDetails} />
            <div className="details">Property details</div>
            <PropertyDetails listing={listing.propertyDetails} />
            <div className="details">Construction details</div>
            <ConstructionDetails listing={listing.constructionDetails} />
            <div className="details">Utilities / Green Energy Details</div>
            <UtilitiesDetails listing={listing.utilitiesGreenEnergyDetails} />
            <div className="details">Community and Neighborhood Details</div>
            <CommunityDetails listing={listing.communityAndNeighborhoodDetails} />
            <div className="details">HOA and financial details</div>
            <HoaDetails listing={listing.hoaAndFinancialDetails} />
            <div className="details">Other</div>
            <Other listing={listing.other} toggleModal={this.toggleModal} />
            <div className="link" onClick={this.expand} tabIndex={0} onKeyPress={this.expand} role="button">
              <img src="https://img.icons8.com/ultraviolet/40/000000/circled-chevron-up.png" alt="" height="20px" width="20px" />
              See less facts and features
            </div>
          </div>
        </div>
      );
    }
    if (expanded === true && modal === false) {
      return (
        <div className="App">
          <h3 className="title-line">Facts and features</h3>
          <GraphicsHeader listing={listing} />
          <div className="details">Interior details</div>
          <InteriorDetails listing={listing.interiorDetails} />
          <div className="details">Property details</div>
          <PropertyDetails listing={listing.propertyDetails} />
          <div className="details">Construction details</div>
          <ConstructionDetails listing={listing.constructionDetails} />
          <div className="details">Utilities / Green Energy Details</div>
          <UtilitiesDetails listing={listing.utilitiesGreenEnergyDetails} />
          <div className="details">Community and Neighborhood Details</div>
          <CommunityDetails listing={listing.communityAndNeighborhoodDetails} />
          <div className="details">HOA and financial details</div>
          <HoaDetails listing={listing.hoaAndFinancialDetails} />
          <div className="details">Other</div>
          <Other listing={listing.other} toggleModal={this.toggleModal} />
          <div className="link" onClick={this.expand} tabIndex={0} onKeyPress={this.expand} role="button">
            <img src="https://img.icons8.com/ultraviolet/40/000000/circled-chevron-up.png" alt="" height="20px" width="20px" />
            See less facts and features
          </div>
        </div>
      );
    }
  }
}

export default App;
