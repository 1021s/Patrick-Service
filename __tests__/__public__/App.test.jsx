/* eslint-env jest */
/* eslint-env node */
import 'jsdom-global/register';
import React from 'react';
import '../../setupTests';
import { mount } from 'enzyme';
import App from '../../public/src/components/App';
import CommunityDetails from '../../public/src/components/CommunityDetails';
import ConstructionDetails from '../../public/src/components/ConstructionDetails';
import GraphicsHeader from '../../public/src/components/GraphicsHeader';
import HoaDetails from '../../public/src/components/HoaDetails';
import InteriorDetails from '../../public/src/components/InteriorDetails';
import PropertyDetails from '../../public/src/components/PropertyDetails';
import UtilitiesDetails from '../../public/src/components/UtilitiesDetails';

describe('App render suite', () => {
  const wrapper = mount(<App />);

  it('should render CommunityDetails without throwing an error', () => {
    const communityDetails = wrapper.contains(CommunityDetails);
    expect(communityDetails).toBe(true);
  });

  it('should render ConstructionDetails without throwing an error', () => {
    const constructionDetails = wrapper.contains(ConstructionDetails);
    expect(constructionDetails).toBe(true);
  });

  it('should render GraphicsHeader without throwing an error', () => {
    const graphicsHeader = wrapper.contains(GraphicsHeader);
    expect(graphicsHeader).toBe(true);
  });

  it('should render HoaDetails without throwing an error', () => {
    const hoaDetails = wrapper.contains(HoaDetails);
    expect(hoaDetails).toBe(true);
  });

  it('should render InteriorDetails without throwing an error', () => {
    const interiorDetails = wrapper.contains(InteriorDetails);
    expect(interiorDetails).toBe(true);
  });

  it('should render PropertyDetails without throwing an error', () => {
    const propertyDetails = wrapper.contains(PropertyDetails);
    expect(propertyDetails).toBe(true);
  });

  it('should render UtilitiesDetails without throwing an error', () => {
    const utilitiesDetails = wrapper.contains(UtilitiesDetails);
    expect(utilitiesDetails).toBe(true);
  });
});

describe('App state suite', () => {
  const wrapper = mount(<App />);
  const state = wrapper.state();

  it('should have state', () => {
    expect(state).not.toBe(undefined);
    expect(state.length).not.toBe(0);
  });
  it('should have a listing state', () => {
    const { listing } = state;
    expect(typeof listing).toBe('object');
  });
  it('should have an expanded state', () => {
    const { expanded } = state;
    expect(typeof expanded).toBe('boolean');
    expect(expanded).toBe(false);
  });
});

describe('App click suite', () => {
  it('button click should hide component', () => {
    const wrapper = mount(<App />);
    wrapper
      .find('.link')
      .simulate('click');

    expect(wrapper).toMatchSnapshot();
  });
});
