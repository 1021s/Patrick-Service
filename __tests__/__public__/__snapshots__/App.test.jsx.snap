// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App click suite button click should hide component 1`] = `ReactWrapper {}`;
